HomeTask4/HomeTask4/ParserForXmlFile.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Text;
+
+namespace HomeTask4
+{
+    /// <summary> 
+    /// This class reads information from the XML file and outputs it to the console
+    /// </summary>
+    class ParserForXmlFile
+    {
+        private string pathToTheFile;
+
+        public ParserForXmlFile(string fileLocation)
+        {
+            pathToTheFile = fileLocation;
+        }
+
+        /// <summary> 
+        /// Method OutputtingDataToConsole
+        /// The method that outputs the generated information
+        /// </summary> 
+        public void OutputtingDataToConsole()
+        {
+            foreach (string nextstring in CreatingArrayOfData())
+            {  
+               Console.WriteLine(nextstring);  
+            }
+        }
+
+        /// <summary> 
+        /// Method CreatingArrayOfData
+        /// A method that processes information from a file
+        /// </summary> 
+        /// <returns>Array of generated information</returns>
+        private string[] CreatingArrayOfData()
+        {
+            string[] informationFromFile = System.IO.File.ReadAllLines(pathToTheFile);
+            string[] arrayOfData = new string[informationFromFile.Length];
+            int counter = 0;
+            StringBuilder way = new StringBuilder();
+            StringBuilder value = new StringBuilder();
```

Comment:
Too complicated and unclear, you have to decompose this method  to a set of classes and methods